js/utils/api.js

```diff
@@ -48,6 +48,37 @@ async function getUserProfile(
   }
 }
 
+async function setUserProfile(
+  apiUrlPrefix: string,
+  token: string,
+  newProfile: object
+) {
+  try {
+    let response = await fetch(`${apiUrlPrefix}/api/v1/profile`, {
+      method: 'post',
+      headers: {
+        Authorization: `Bearer ${token}`,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        email: newProfile.email,
+        is_inbox_enabled: newProfile.is_inbox_enabled,
+        version: newProfile.version
+      })
+    })
+
+    if (response.status == 500) {
```

Comment:
there can be other error codes returned, see specs

Reply:
Same here.